tests/test_spatial_data_errors.py

```
@@ -95,3 +101,144 @@ def test_overpass_http_error_status_returns_none(self):
+            second = discovery.get_year_directory_contents(2018)
+            assert first == second
+            assert mock_get.call_count == 1, "second call must hit the cache, not the network"
+
+    def test_429_fallback_only_for_skip_strategy(self):
+        """Callers with on_error='raise' must still receive the HTTPError."""
+        discovery = CensusDirectoryDiscovery()
+        with patch(
+            "siege_utilities.geo.spatial_data.requests.get",
+            side_effect=self._make_429_error(),
+        ):
+            with pytest.raises(Exception):
```

Ruff (0.15.11)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)